maintenance/GenerateMissingIdentifiers.php

```diff
@@ -20,7 +21,49 @@ public function __construct() {
 	}
 
 	public function execute(): void {
-		$this->output( 'TODO' );
+		$generatedIdsCount = 0;
+
+		while ( true ) {
+			$pageIds = $this->getNextBatchOfPageIdsForPagesWithoutPersistentIds();
+			$batchCount = count( $pageIds );
+
+			if ( $batchCount === 0 ) {
+				break;
+			}
+
+			$this->output( "Generating persistent ids for batch of $batchCount pages\n" );
+
+			$this->savePersistentIds( $pageIds, $this->generatePersistentIds( $batchCount ) );
+
+			$generatedIdsCount += $batchCount;
+		}
+
+		$this->output( "Generated $generatedIdsCount persistent IDs\n" );
+	}
+
+	/**
+	 * @return int[]
+	 */
+	private function getNextBatchOfPageIdsForPagesWithoutPersistentIds(): array {
+		return PersistentPageIdentifiersExtension::getInstance()->getPageIdsRepo()
+			->getPageIdsOfPagesWithoutPersistentIds( limit: 100 );
```

Comment:
Just an arbitrary batch size for now. I have no idea of the time/performance implications.

Reply:
Bigger batch should be better performance, until the point where we hit some memory or similar limits, and the write query fails. I suspect we can do 1000 without issues given we are dealing with integers